src/libraries/Common/tests/System/Xml/XmlDiff/XmlDiffDocument.cs

```diff
@@ -1550,15 +1560,19 @@ public class XmlDiffCharacterData : XmlDiffNode
     {
         private string _value;
         private XmlDiffNodeType _nodetype;
-        public XmlDiffCharacterData(string value, XmlDiffNodeType nt, bool NormalizeNewline)
+        public XmlDiffCharacterData(string value, XmlDiffNodeType nt, bool NormalizeNewline, bool NormalizeSpaces, bool IgnoreWhitespace)
             : base()
         {
-            this._value = value;
             if (NormalizeNewline)
-            {
-                this._value = this._value.Replace("\n", "");
-                this._value = this._value.Replace("\r", "");
-            }
+                value = value.Replace("\n", "").Replace("\r", "");
+
+            if (NormalizeSpaces)
+                value = Regex.Replace(value, "[\u00A0\u180E\u2000-\u200B\u202F\u205F\u3000\uFEFF]", " ");
```

Comment:
> value = Regex.Replace(value, "[\u00A0\u180E\u2000-\u200B\u202F\u205F\u3000\uFEFF]", " "); [](http://example.com/codeflow?start=15&length=90)

would it be better to move this to a helper method instead of duplicating it?

Reply:
Certainly, or a Regex member, or at least a constant :) Good call.

What's the preference for this sort of thing in test classes?